Procedure
=========

.. important::
   This section is a fundamental building block of any SOP
   
The 'Procedure' section of an SOP should provide a detailed, step-by-step description of how to carry out the specific tasks or processes outlined in the SOP. This section should be clear, concise, and easy to follow, ensuring that personnel can perform the procedure consistently and accurately.

The content of the procedure section will vary considerably depending on the nature of the SOP.

-----

.. rubric:: Example content:

.. dropdown:: üß¨ Bioinformatics QC Procedure

   *Example SOP content for a bioinformatics QC pipeline*
   
   .. rubric:: Procedure
   
   **Receipt of sequencing files**
   
   Sequencing files are automatically imported onto the staging server upon completion of an Illumina sequencing run. The bioinformatics production server checks the staging server for completed runs every 10 minutes via a scheduled cron job. When a new completed run is detected, the production server automatically imports the files from the staging server to `/mnt/data/raw/<SEQUENCER_ID>/<RUN_ID>/` and initiates the bioinformatics QC pipeline, which is implemented in Nextflow and maintained on `GitHub <https://github.com/lab-bioinformatics/qc-pipeline>`_.

   **Executing the analysis**
   
   The bioinformatics QC pipeline can be executed in two modes: 

   1.	**Automated Mode:** The pipeline is launched automatically when a new sequencing run is copied from the staging server.
   2.	**Manual Mode:** A user can manually initiate the pipeline on a selected set of FASTQ files.

   **Manual Mode**

   .. note::
      Manual mode can only be initiated by a member of the bioinformatics team as access to servers is restricted.
   
   Manual mode should only be used where a run has failed because of a server issue (e.g. a power cut that stopped the analysis part way through) or because sequence data needs to be imported from another source (e.g. where fastq files have been generated by a collaborating organisation and we wish to be able to analyse the locally within the laboratory.

   To restart a failed run:

   Log into the production server using an ssh client and switch to a user profile with appropriate permissions:

   .. code-block:: bash

      ssh username@10.0.1.12
      sudo su - prod-user




   






.. dropdown:: üë©‚Äçüî¨ Staff Training Procedure

   *Example SOP content for a training procedure*

   .. rubric:: Procedure
   
   Add content

.. dropdown:: üß™ Laboratory Procedure

   *Example SOP content for a nucleic acid extraction procedure*

   .. rubric:: Procedure
   
   Add content

.. dropdown:: üíª Code Update & Review Procedure

   *Example SOP content for a code update and review procedure*

   .. rubric:: Procedure
   
   Add content

.. dropdown:: ‚úÖ Verification Procedure

   *Example SOP content for a verification procedure*

   .. rubric:: Procedure
   
   Add content

.. raw:: html

   <script>
     // Auto-close other dropdowns when one opens
     document.querySelectorAll('details').forEach((el) => {
       el.addEventListener('toggle', function () {
         if (el.open) {
           document.querySelectorAll('details').forEach((other) => {
             if (other !== el) {
               other.removeAttribute('open');
             }
           });
         }
       });
     });
   </script>
